core/modules/recursive_load.rs

```diff
@@ -265,14 +266,17 @@ impl RecursiveModuleLoad {
               if visited_as_alias.borrow().contains(specifier.as_str()) {
                 return Ok(None);
               }
-              let load_result = loader
-                .load(
-                  &specifier,
-                  Some(&referrer),
-                  is_dynamic_import,
-                  requested_module_type,
-                )
-                .await;
+              let load_response = loader.load(
+                &specifier,
+                Some(&referrer),
+                is_dynamic_import,
+                requested_module_type,
+              );
+
+              let load_result = match load_response {
+                ModuleLoadResponse::Sync(result) => result,
+                ModuleLoadResponse::Async(fut) => fut.await,
+              };
```

Comment:
This is the critical file; check if having sync response can cause some optimizations here.